hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/keyvalue/FileReadableKVState.java

```diff
@@ -34,26 +36,36 @@ public class FileReadableKVState extends AbstractReadableKVState<FileID, File> {
     private final FileDataRepository fileDataRepository;
     private final EntityRepository entityRepository;
     private final SystemFileLoader systemFileLoader;
+    private final CommonProperties commonProperties;
 
     public FileReadableKVState(
             final FileDataRepository fileDataRepository,
             final EntityRepository entityRepository,
-            SystemFileLoader systemFileLoader) {
+            SystemFileLoader systemFileLoader,
+            CommonProperties commonProperties) {
         super(KEY);
         this.fileDataRepository = fileDataRepository;
         this.entityRepository = entityRepository;
         this.systemFileLoader = systemFileLoader;
+        this.commonProperties = commonProperties;
     }
 
     @Override
     protected File readFromDataSource(@Nonnull FileID key) {
         final var timestamp = ContractCallContext.get().getTimestamp();
         final var fileEntityId = toEntityId(key);
         final var fileId = fileEntityId.getId();
+        final var currentTimestamp = getCurrentTimestamp();
+
+        if (SystemEntity.THROTTLE_DEFINITIONS
+                .getScopedEntityId(commonProperties)
+                .equals(fileEntityId)) {
+            return systemFileLoader.loadThrottles(fileId, key, currentTimestamp);
```

Comment:
I think what we want is for `SystemLoader` to take care of loading system files from db or classpath. With current change some methods load from db while others only load from classpath as a fallback. I think we should take the time now to refactor it to make `load()` load from db and be done first and not as a fallback for all system files. This also avoids having to check fileID in both kv state and in loader.

```suggestion
        var systemFile = systemFileLoader.load(key, currentTimestamp);
        if (systemFile != null) {
            return systemFile;
        }
```

Reply:
Me might want to avoid this due to concerns described here https://github.com/hiero-ledger/hiero-mirror-node/pull/10687#discussion_r2010600147